components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -312,59 +312,76 @@ private Response processRequest(final Map<String, String> requestAttributes) {
 
-                scimResponse = groupResourceManager.listWithGET(userManager, filter, startIndex,
-                        count, sortBy, sortOrder, domainName, attributes, excludedAttributes);
-
+                // Processing count and startIndex in the request.
+                Integer startIndex = convertStringPaginationParamsToInteger(
+                        requestAttributes.get(SCIMProviderConstants.START_INDEX), SCIMProviderConstants.START_INDEX);
+                Integer count = convertStringPaginationParamsToInteger(requestAttributes.get(SCIMProviderConstants.COUNT),
+                        SCIMProviderConstants.COUNT);
+                scimResponse = groupResourceManager
+                        .listWithGET(userManager, filter, startIndex, count, sortBy, sortOrder, domainName, attributes,
+                                excludedAttributes);
             } else if (GET.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.get(id, userManager, attributes, excludedAttributes);
-            } else if (POST.class.getSimpleName().equals(httpVerb) && search.equals("1")){
-                scimResponse = groupResourceManager.listWithPOST(resourceString,userManager);
+            } else if (POST.class.getSimpleName().equals(httpVerb) && search.equals("1")) {
+                scimResponse = groupResourceManager.listWithPOST(resourceString, userManager);
             } else if (POST.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.create(resourceString, userManager, attributes, excludedAttributes);
             } else if (PUT.class.getSimpleName().equals(httpVerb)) {
-                scimResponse =
-                        groupResourceManager.updateWithPUT(id, resourceString, userManager, attributes, excludedAttributes);
+                scimResponse = groupResourceManager
+                        .updateWithPUT(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (PATCH.class.getSimpleName().equals(httpVerb)) {
-                scimResponse =
-                        groupResourceManager.updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
+                scimResponse = groupResourceManager
+                        .updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (DELETE.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.delete(id, userManager);
             }
-
             return SupportUtils.buildResponse(scimResponse);
-
         } catch (CharonException e) {
             return handleCharonException(e, encoder);
         }
     }
 
+    /**
+     * Method to convert string pagination values to Interger pagination values.
+     *
+     * @param valueInRequest       Value passed in the request.
+     * @param scimProviderConstant The name of parameter (count or startIndex).
```

Comment:
Shall we remove `count or startIndex` ?

Reply:
Fixed: 2a9df39b802d0a818f4c26bdd0f6c7ffe48a4136